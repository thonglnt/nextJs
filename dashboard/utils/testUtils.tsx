import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
