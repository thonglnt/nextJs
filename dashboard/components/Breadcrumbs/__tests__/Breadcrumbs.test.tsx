// Utils
import { render, screen } from '@utils/testUtils'

// Components
import BasicBreadcrumbs, { IBreadcrumbs } from '..'

describe('Breadcrumbs component', () => {
  const mockProps: IBreadcrumbs = {
    title: 'Products',
  }

  it('should render Breadcrumb component', () => {
    const { container } = render(<BasicBreadcrumbs {...mockProps} />)

    expect(container).toMatchSnapshot()
  })

  it('should render match content Breadcrumb component', () => {
    render(<BasicBreadcrumbs {...mockProps} />)

    const dashBoard = screen.getByText(/Dashboard/i)
    const management = screen.getByText(/Management/i)
    const products = screen.getByText(/Products/i)

    expect(dashBoard).toHaveTextContent('Dashboard')
    expect(management).toHaveTextContent('Management')
    expect(products).toHaveTextContent('Products')
  })
})
