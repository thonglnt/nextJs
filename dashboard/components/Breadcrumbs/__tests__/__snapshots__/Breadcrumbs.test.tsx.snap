// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component should render Breadcrumb component 1`] = `
<div>
  <nav
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-i4iwuk-MuiTypography-root"
        >
          <a
            href="/"
          >
            Dashboard
          </a>
        </p>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-i4iwuk-MuiTypography-root"
        >
          <a
            href="/"
          >
            Management
          </a>
        </p>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-8vdomm-MuiTypography-root"
        >
          Products
        </p>
      </li>
    </ol>
  </nav>
</div>
`;
